src/coreclr/tools/aot/ILCompiler/ILCompiler.csproj

```diff
@@ -10,6 +10,7 @@
   <!-- BEGIN: Workaround for https://github.com/dotnet/runtime/issues/67742 -->
   <PropertyGroup Condition="'$(BuildingInsideVisualStudio)' != 'true'">
     <PublishDir>$(RuntimeBinDir)ilc-published/</PublishDir>
+    <NativeAotSupported Condition="'$(RuntimeIdentifier)' != 'win-x64' and '$(RuntimeIdentifier)' != 'linux-x64'">false</NativeAotSupported>
```

Comment:
Why exclude macOS (and arm64)?

Reply:
See the previous failed attempts: `Directory.Build.targets(83,5): error : (NETCORE_ENGINEERING_TELEMETRY=Build) The passed-in TargetOS property value 'OSX' must be lowercase.`

Directory.Build.targets became opinionated about how to spell OSX. The old ilc probably doesn't spell it right. I don't particularly care about osx in the scope of this PR - this is to speed up testing and we do the majority of testing on Linux and Windows.

Reply:
I think we can broaden this pretty easily after we get it started and flush out the problems.